data_model/src/trigger/set.rs

```diff
@@ -120,14 +120,61 @@ impl Set {
         self.ids.iter().map(|entry| entry.key().clone()).collect()
     }
 
+    /// Apply `f` to triggers that belong to the given [`DomainId`]
+    ///
+    /// Returns an empty list if [`Set`] doesn't contain any trigger belonging to [`DomainId`].
+    pub fn inspect_by_domain_id<F, R>(&self, domain_id: &DomainId, f: F) -> Vec<R>
+    where
+        F: Fn(&Id, Action<FilterBox>) -> R,
+    {
+        self.ids
+            .iter()
+            .filter_map(|pair| {
+                let id = pair.key();
+                let event_type = pair.value();
+
+                let trigger_domain_id = id.domain_id.as_ref()?;
+
+                let action = match event_type {
+                    EventType::Data => self
+                        .data_triggers
+                        .get(id)
+                        .map(|entry| ActionTrait::clone_and_box(entry.value()))
+                        .expect("`Set::data_triggers` doesn't contain required id. This is a bug"),
+                    EventType::Pipeline => self
+                        .pipeline_triggers
+                        .get(id)
+                        .map(|entry| ActionTrait::clone_and_box(entry.value()))
+                        .expect(
+                            "`Set::pipeline_triggers` doesn't contain required id. This is a bug",
+                        ),
+                    EventType::Time => self
+                        .time_triggers
+                        .get(id)
+                        .map(|entry| ActionTrait::clone_and_box(entry.value()))
+                        .expect("`Set::time_triggers` doesn't contain required id. This is a bug"),
+                    EventType::ExecuteTrigger => self
+                        .by_call_triggers
+                        .get(id)
+                        .map(|entry| ActionTrait::clone_and_box(entry.value()))
+                        .expect(
+                            "`Set::by_call_triggers` doesn't contain required id. This is a bug",
+                        ),
+                };
+
+                (trigger_domain_id == domain_id).then(|| f(id, action))
```

Comment:
This looks elegant, but I think it's better to place this check before `match event_type` to not to search for trigger that won't be executed